src/components/catalog-search/catalog-search.controller.ts

```diff
@@ -72,6 +80,29 @@ export class CatalogSearchController implements angular.IController {
     return this.filterForKeywords(searchText);
   }
 
+  public toggleMobileShowSearchInput() {
+    this.ctrl.mobileSearchInputShown = !this.ctrl.mobileSearchInputShown;
+    this.ctrl.searchText = '';
+    if (_.isFunction(this.ctrl.searchToggleCallback)) {
+      this.ctrl.searchToggleCallback(this.ctrl.mobileSearchInputShown);
+    }
+    if (this.ctrl.mobileSearchInputShown) {
+      this.setSearchInputFocus(0);
+    }
+  }
+
+  private setSearchInputFocus(timeSoFar: number) {
```

Comment:
Does the `take-focus` directive work?

https://github.com/openshift/origin-web-common/blob/7b835d9f76f3ade1f0366213e8b7564fee2238e7/src/components/take-focus/takeFocus.js

Reply:
No, I tried that first but it doesn't.

Reply:
take-focus works when you first add a component. Since this is just hidden and shown based on the size and the toggle it won't work here.

We could use ng-if to hide and show it but that would mean having the input (with all 15 parameters) duplicated for mobile mode and also watch for window resizes to determine which input to show.

Or, we could add this kind of functionality to the take-focus directive so that a component gets focus when it is added or when it becomes visible.